### PR TITLE
refactor: rename exception replay enablement variable

### PR DESCRIPTION
**Pull request by P403n1x87**
We rename the exception replay enablement variable from `DD_EXCEPTION_DEBUGGING_ENABLED` to `DD_EXCEPTION_REPLAY_ENABLED`

Requires #9724.

## Checklist

- [x] The PR description includes an overview of the change
- [x] The PR description articulates the motivation for the change
- [x] The change includes tests OR the PR description describes a testing strategy
- [x] The PR description notes risks associated with the change, if any
- [x] Newly-added code is easy to change
- [x] The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- [x] The change includes or references documentation updates if necessary
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate
- [x] All changes are related to the pull request's stated goal
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- [x] Testing strategy adequately addresses listed risks
- [x] Newly-added code is easy to change
- [x] Release note makes sense to a user of the library
- [x] If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Comment by P403n1x87** [comment]
We need to upgrade `envier` to merge this

**Review by shatzi — APPROVED** [review]

**Review by mabdinur — APPROVED** [review]

File: releasenotes/notes/refactor-di-er-enablement-env-var-12056782561eecc6.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+deprecations:
+  - |
+    exception replay: The ``DD_EXCEPTION_DEBUGGING_ENABLED`` environment
+    variable has been deprecated in favor of ``DD_EXCEPTION_REPLAY_ENABLED``.
+    The old environment variable will be removed in a future major release.
```
Comment (erikayasuda) [inline releasenotes/notes/refactor-di-er-enablement-env-var-12056782561eecc6.yaml:6]: Should we explicitly specify 3.x here?
```suggestion
    The old environment variable will be removed in the next major release (3.x).
```

File: tests/debugging/exception/test_replay.py [hunk 2]
```diff
@@ -3,36 +3,56 @@
...
+
+    er_config = ExceptionReplayConfig()
+    assert er_config.enabled
+
+
+def test_exception_replay_config_enabled_deprecated(monkeypatch):
+    monkeypatch.setenv("DD_EXCEPTION_DEBUGGING_ENABLED", "1")
+
+    er_config = ExceptionReplayConfig()
+    assert er_config.enabled
```
Comment (erikayasuda) [inline tests/debugging/exception/test_replay.py:25]: I'm assuming this is some envier magic that allows users to still use `DD_EXCEPTION_DEBUGGING_ENABLED` until it's unsupported in 3.x?

File: ddtrace/settings/exception_replay.py [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
+    __prefix__ = "dd.exception"
 
     enabled = En.v(
         bool,
-        "enabled",
+        "replay.enabled",
         default=False,
         help_type="Boolean",
         help="Enable automatic capturing of exception debugging information",
+        deprecations=[("debugging.enabled", None, "3.0")],
```
Comment (erikayasuda) [inline ddtrace/settings/exception_replay.py:13]: Does this also log the deprecation warning to the customer + allow them to use the legacy env var until 3.0 is released?

**Review by erikayasuda — APPROVED** [review]
🫨

File: tests/debugging/exception/test_replay.py (hunk 2 above)
Comment (P403n1x87) [reply tests/debugging/exception/test_replay.py:25]: Yes, if the actual variable is not specified, envier looks for the `deprecations`. If any are set, they are used instead and a deprecation warning is emitted

File: ddtrace/settings/exception_replay.py (hunk 3 above)
Comment (P403n1x87) [reply ddtrace/settings/exception_replay.py:13]: Yep 🙂

**Review by brettlangdon — APPROVED** [review]

**Review by gnufede — APPROVED** [review]

**Review by romainkomorndatadog — APPROVED** [review]

**Pull request merged**